src/lib/mathlib/math/Functions.hpp

```diff
@@ -60,11 +60,12 @@ int sign(T val)
  * 		1 - pure cubic function
  * @return result of function output
  */
-template<typename _Tp>
-inline const _Tp expo(const _Tp &value, const _Tp &e)
+template<typename T>
+const T expo(const T &value, const T &e)
 {
-	_Tp x = constrain(value, (_Tp) - 1, (_Tp)1);
-	return (1 - e) * x + e * x * x * x;
+	T x = constrain(value, (T) - 1, (T) 1);
+	T ec = constrain(e, (T) 0, (T) 1);
```

Comment:
Since the API clearly specifies the valid range already, I don't see this is needed - unnecessary computation. But if you feel it's error prone otherwise, you can leave it in.

Reply:
I did it for safety because if the parameters are set wrong the drone might get uncontrollable by stick.

Reply:
Ok, but then the check should be done when loading the parameters, not deeply inside a math routine, which is called at high frequency.

Reply:
You're totally right. 😞